base/applications/rapps/loaddlg.cpp

```diff
@@ -710,8 +745,12 @@ CDownloadManager::ThreadFunc(LPVOID param)
                 Path += APPLICATION_DATABASE_NAME;
                 break;
             case DLTYPE_APPLICATION:
-                Path += (LPWSTR)(p + 1); // use the filename retrieved from URL
+            {
+                CStringW str = p + 1; // use the filename retrieved from URL
```

Comment:
Just to be sure, is the CStringW constructor here making a _**copy**_ (memory allocation and characters copy) of the string pointed by `p + 1` ? Or is the str internally still pointing to the original buffer pointed by  `p + 1`  ?

Reply:
This `p` is pointing to the last slash in a URL.
`p + 1` will be the next character of the last slash.